### DISCUSSION TITLE
Keymap doesn't overwrite default mapping

### DISCUSSION THREAD

**Discussion by pfiadDi** [neovim/neovim#30003] [Q&A] [answered]
Hi, in my init.lua I have this mapping:

`vim.keymap.set('i', '<C-i>', '<Esc>', { desc = 'Back to normal mode', remap = true })
`
But it doesn't work, `<C-i>` still resolves in `<Tab>` as it is it's default.
I tried others, but if they are taken my mapping doesn't apply. For now I use `<C-z>` that works but I don't like it. 

What am I doing wrong? 
Any tips / ideas? 
Thanks

**Comment by zeertzjq** [comment]
It's possible that you have a plugin that maps `<Tab>` in Insert mode. Then your mapping won't work if your terminal sends the same sequences for `<C-I>` and `<Tab>`.

**Reply by pfiadDi** [reply]
Thank you for your answer. No I don't have this mapping. As I said, `<C-i>` is the default mapping in insert mode for `<Tab>`. But can't I overwrite defaults?

EDIT: I also checked the plugins. none is setting `<C-i>` to `<Tab>`. That's just the default settings according to the defaults in :help index

**Reply by zeertzjq** [reply]
I was suggesting that you check if there is mapping that maps `<Tab>`, not one that maps `<C-I>`. And what's your terminal emulator?

**Reply by pfiadDi** [reply]
Yes! Thank you. I found it. The autocomplete plugin cmp uses `<Tab>`. 
I wanted to use `<Tab>` to select the auto sugg:

`
          ['<Tab>'] = cmp.mapping.confirm { select = true },
`
 
I don't understand, why this is happening. Can you explain it to me?
Default: `<C-i>` maps to `<Tab>` in insert mode
My config:
`<Tab> = cmp.mapping.confirm { select = true }` -> in cmp.setup
and below in init.lua:
`vim.keymap.set('i', '<C-i>', '<Esc>', { desc = 'Back to normal mode' })`

So I don't understand, why `<C-i>` still maps to `<Tab>` when I added a new mapping.
EDIT: I removed remap = true now, since I understand now what it does

**Reply by zeertzjq** [reply]
What's your terminal emulator?

**Reply by pfiadDi** [reply]
I use the normal bash terminal on Ubuntu

**Reply by zeertzjq** [reply, marked as answer]
GNOME Terminal sends the same sequences for `<C-I>` and `<Tab>`, so your `<C-I>` mapping won't work if you have another mapping for `<Tab>`. You can try a terminal emulator that supports modifyOtherKeys or CSI u, like WezTerm, XTerm, kitty or foot.

**Reply by pfiadDi** [reply]
Thank you so much that you took the time to help and explain everything to me!
